packages/@lwc/engine-dom/src/apis/build-custom-element-constructor.ts

```diff
@@ -47,9 +47,12 @@ export function deprecatedBuildCustomElementConstructor(
     return Ctor.CustomElementConstructor;
 }
 
-// Note: WeakSet is not supported in IE11, and the polyfill is not performant enough.
-//       This WeakSet usage is valid because this functionality is not meant to run in IE11.
-const hydratedCustomElements = new WeakSet<Element>();
+function clearNode(node: Node) {
+    const childNodes = renderer.getChildNodes(node);
+    for (let i = 0; i < childNodes.length; i++) {
```

Comment:
```suggestion
    for (let i = childNodes.length -1 ; i >= 0; i--) {
```

Otherwise, I believe this will not actually remove all the nodes, since the child nodes length is changing while the array is being mutated.